src/test/consensus/Consensus_test.cpp

```diff
@@ -522,6 +522,162 @@ class Consensus_test : public beast::unit_test::suite
+
+            NetClock::duration when = sim.peers[0].now().time_since_epoch();
+
+            // Check we are before the 30s to 20s transition
+            NetClock::duration resolution =
+                sim.peers[0].lastClosedLedger.closeTimeResolution();
+            BEAST_EXPECT(resolution == NetClock::duration{30s});
+
+            while (
+                ((when % NetClock::duration{30s}) != NetClock::duration{15s}) ||
+                ((when % NetClock::duration{20s}) != NetClock::duration{15s}))
+                when += 1s;
+            // Advance the clock without consensus running (IS THIS WHAT
+            // PREVENTS IT IN PRACTICE?)
+            sim.net.step_for(
+                NetClock::time_point{when} - sim.peers[0].now());
+
+            // Run one more ledger with 30s resolution
+            sim.run(1);
+
+            // close time should be ahead of clock time since we engineered
+            // the close time to round up
+            for (Peer const & peer : sim.peers)
+            {
+                BEAST_EXPECT(
+                    peer.lastClosedLedger.closeTime() > peer.now());
+                BEAST_EXPECT(peer.lastClosedLedger.closeAgree());
+                BEAST_EXPECT(
+                    peer.lastClosedLedger.id() ==
+                    sim.peers[0].lastClosedLedger.id());
+            }
+
+
+            // All peers submit their own ID as a transaction
+            for (Peer & peer : sim.peers)
+                peer.submit(Tx{static_cast<std::uint32_t>(peer.id)});
+
+            // Run 1 more round, this time it will have a decreased
+            // resolution of 20 seconds.
+            // 86481
+
+            // The network delays are engineered so that the slow peers
+            // initially have the wrong tx hash, but they see a majority
+            // of agreement from their peers and declare consensus
+            //
+            // The trick is that everyone starts with a raw close time of
+            //  84681s
+            // Which has
+            //   effCloseTime(86481s, 20s,  86490s) = 86491s
+            // However, when the slow peers update their position, they change
+            // the close time to 86451s. The fast peers declare consensus with
+            // the 86481s as their position still.
+            //
+            // When accepted the ledger
+            // - fast peers use eff(86481s) -> 86491s as the close time
+            // - slow peers use eff(eff(86481s)) -> eff(86491s) -> 86500s!
+
+            sim.run(1);
+
+            if (parms.useRoundedCloseTime)
+            {
+                for (Peer const& peer : sim.peers)
+                {
+                    BEAST_EXPECT(
+                        peer.lastClosedLedger.id() ==
+                        sim.peers[0].lastClosedLedger.id());
+                }
+            }
+            else
+            {
+                // All fast peers agreed on LCL TX id
+                BEAST_EXPECT(std::all_of(
```

Comment:
isn't this check effectively the same as the one starting on line 635, so code could be DRY'd?

Reply:
Good catch, I'll fix. For context, these changes were actually back ported from a branch based off of #2209 where the comparisons need to be different.